src/v/cluster/partition_balancer_planner.cc

```diff
@@ -1065,15 +1067,15 @@ partition_balancer_planner::plan_data partition_balancer_planner::plan_actions(
     }
 
     if (
-      result.violations.is_empty()
+      result.violations.is_empty() && ctx.decommissioning_nodes.empty()
```

Comment:
We shouldn't skip decommissioning even if maintenance mode is on. It should still be a priority